### PR TITLE
Build a better cache for the CompactingChest

### LINKED ISSUE #38: [BUG] Compacting chest causes massive tps lagg
**Describe the bug**
On the AOF5 server I play we had big problems with tps lagg, so we used spark to run a profiler and one compacting chest had the impact of 1.6% of the whole server ticktime. Nearly all of this time is happening in the recipemanager call.
This is happening because the cache is forgotten after each non successful crafting request, meaning it needs to look through the recipes again. Also if you have for example more than the required compacting items of something non compactable in the first slots it will combine those slots multiple times and check if a valid recipe for them can be found multiple times, causing more lagg.

I've created a better, shared cache solution in a PR coming soon, so this issue is kinda just a documentation of this problem

**To Reproduce**
Load a datapack with a large amount of crafting recipes and slowly fill the compacting chest using a hopper with new, non compressable items, causing the compacting chest to look for valid recipes over all item slots.

**Expected behavior**
The server not to lagg

**Mod version**
This issue was found in 1.2.8-1.18.1-midgame9, but I could also verify it in the latest github commit.

### PR DESCRIPTION
**Pull request by justliliandev**
Also improved the reaction time to automatic filling of the chest by trying to craft every second.

Closes #38

### CODE REVIEW

**Comment by DaFuqs** [comment]
Awesome! I will have a look at it

**Comment by justliliandev** [comment]
If you have any question, don't hesitate to ping me on discord

File: src/main/java/de/dafuqs/spectrum/blocks/chests/CompactingChestBlockEntity.java [hunk 1]
```diff
@@ -206,4 +224,29 @@ public void writeScreenOpeningData(ServerPlayerEntity player, PacketByteBuf buf)
...
+			if (o == null || getClass() != o.getClass()) return false;
+			InputItem that = (InputItem) o;
+			return item.equals(that.item) && Objects.equals(nbt, that.nbt);
+		}
+
+		@Override
+		public int hashCode() {
+			return Objects.hash(item, nbt);
+		}
+	}
```
Comment (BasiqueEvangelist) [inline src/main/java/de/dafuqs/spectrum/blocks/chests/CompactingChestBlockEntity.java]: Why not use Fabric API's ItemVariant here?

Comment (justliliandev) [reply src/main/java/de/dafuqs/spectrum/blocks/chests/CompactingChestBlockEntity.java]: I'm usually modding with forge and didn't knew about the existence of ItemVariant

Comment (justliliandev) [reply src/main/java/de/dafuqs/spectrum/blocks/chests/CompactingChestBlockEntity.java]: I've updated the PR to use ItemVariant

**Comment by DaFuqs** [comment]
looking good!
I will release an updated version as soon as I figured out the best way to handle the current Origins incompatibility

**Pull request merged**